hazelcast-sql/src/main/java/com/hazelcast/sql/impl/schema/Mapping.java

```diff
@@ -106,8 +107,12 @@ public List<MappingField> fields() {
         return Collections.unmodifiableList(mappingFields);
     }
 
+    /**
+     * Returned options are modifiable because {@linkplain KvMetadataResolver#getSchemaId
```

Comment:
see https://github.com/hazelcast/hazelcast/pull/25269/commits/ea954af7a05e9b73a1041b2ab96e5c0e1f37e6a7#r1360741869

Reply:
I finally managed to propagate the preferred schema to `HazelcastKafkaAvro[De]Serializer` without making mapping options mutable: `KafkaTable` passes the schemas carried by key/value `UpsertTargetDescriptor`s to `PropertiesResolver`, where they are put into consumer/producer properties.